jekyll/_cci2/config-policy-management-overview.adoc

```diff
@@ -361,6 +361,224 @@ disallow_ssh_rerun = "Cannot perform ssh_rerun with sensitive contexts" {
+
+Writing policies for CircleCI `version 2.1` configuration introduces some challenges due to the parameterization and reusable configuration options. To read more about these options, see the xref:reusing-config#[Reusable config reference guide].
+
+Before executing any pipelines, config `version 2.1` is compiled into config `version 2.0`. This compilation expands all parameters and reusable config blocks (jobs, executors, commands, orbs) into workflows and jobs.
+
+To write highly effective policies, it is essential to reference the _compiled_ version of the config (`input._compiled_`).
+
+[#example]
+=== Example
+Consider the following example policy and configuration:
+
+**Policy**
+[source,rego]
+----
+import future.keywords
+
+policy["name"] = "example_mistake"
+
+enable_hard["enforce_not_large_resource"]
+
+# check every job in input config, and if any job has resource_class equal to "large" set a violation message.
+enforce_not_large_resource[reason] {
+  some job_name, job in input.jobs
+  job.resource_class == "large"
+  reason = sprintf("job %s using banned large resource class", [job_name])
+}
+----
+
+**Configuration with reusable executor**
+[source,yaml]
+----
+version: 2.1
+
+executors: # Define reusable executor
+  lg-executor:
+    docker:
+      - image: my-image
+    resource_class: large # Resource class configured in reusable executor
+
+jobs:
+  test:
+    executor: lg-executor
+----
+
+In the above example, the policy is bypassed and will **not trigger**. The policy inspects `jobs` and does not find a `resource_class == "large"`.
+
+This is problematic because once the configuration is compiled, the job `test` will have a `resource_class == "large"`.
+
+Another way this policy could be unintentionally bypassed is by using parameters. Consider the following configuration, which uses a parameter to set the resource class for an executor:
+
+**Configuration using a parameter**
+[source,yaml]
+----
+version: 2.1
+
+jobs:
+  test:
+    parameters:
+      size:
+        type: string
+    resource_class: << parameters.size >> # parameterized definition of resource_class
+
+workflows:
+  main:
+    jobs:
+      - test: # invokation of parameterized job "test" with a size equal to "large".
+          size: large
+----
+
+The same situation applies as for the first configuration presented above. The policy inspects the jobs and does not find a `resource_class == "large"`, but instead finds `<< parameters.size >>`, which is acceptable for the policy.
+
+However, once the config is compiled, the job test will have a `resource_class == "large"`.
+
+To resolve both of these issues, it is important to acknowledge that we want to apply the policy to all jobs, which is a configuration `version: 2.0` construct, and write the policy to target the compiled version accordingly, as follows:
```

Comment:
Non-blocking for this PR, just something to consider: I think it might be helpful if we included some context about what config `version: 2.0` means, and maybe also what we mean when we say "construct". Not sure that we have any existing page or section somewhere that does this. Could it also possibly be good to have this sort of info in a snippet?

Reply:
Yes excellent suggestion. I will make a ticket for this ❤️